src/adam_core/coordinates/tests/test_covariances.py

```diff
@@ -44,9 +45,25 @@ def test_sample_covariance_sigma_points():
         np.testing.assert_allclose(covariance_sg, covariance, rtol=0, atol=1e-14)
 
 
+def test_make_positive_semidefinite():
```

Comment:
Can we get a test of `sample_covariance_random` that verifies the tolerance is working as expected? One where it fixes it within the tolerance and one that throws the error since it's outside the tolerance?